project/Dependencies.scala

```diff
@@ -9,7 +9,7 @@ object Dependencies {
     val scala212 = "2.12.8"
     val scala213 = "2.13.0"
 
-    val akka = "2.5.25"
+    val akka = "2.6.0-M7"
```

Comment:
I'm not sure we should do this (yet). It means that next release will not work with Akka 2.5.x (and we haven't even release 2.6.0 yet).

Was there something specific that didn't work?

Exactly when to switch the dependency can be discussed, but it will mean that we have to maintain two branches and if we possible we should avoid that (for a while).

Reply:
When using Lagom `1.6.0-M6` in https://github.com/lagom/lagom-samples/pull/99#pullrequestreview-295416315 we noticed the `akka-grpc` and/or `play-grpc` code generators produce code with a handful of deprecation warnings. 

The warning are not only for the use of `ActorMaterializer` from the code generated by akka-grpc, but also in gRPC-specific code using deprecated API.

I wanted to reduce the number of moving parts to understand better the issues. 

I just realised the use of protobuf v3 in Akka 2.6.0-M8 may be contributing to this collection of deprecation warnings.

Here's a snippet:

```

[warn] /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/grpc-example/grpc-example-java/hello-proxy-impl/target/scala-2.12/src_managed/main/example/myapp/helloworld/grpc/GreeterServiceClient.java:55:1: akka.stream.ActorMaterializer in akka.stream has been deprecated
[warn]         if (mat instanceof ActorMaterializer) {
[warn] /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/grpc-example/grpc-example-java/hello-proxy-impl/target/scala-2.12/src_managed/main/example/myapp/helloworld/grpc/GreeterServiceClient.java:56:1: akka.stream.ActorMaterializer in akka.stream has been deprecated
[warn]           ((ActorMaterializer) mat).system().getWhenTerminated().whenComplete((v, e) -> close());
[warn] /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/grpc-example/grpc-example-java/hello-proxy-impl/target/scala-2.12/src_managed/main/example/myapp/helloworld/grpc/HelloWorldProto.java:43:1: com.google.protobuf.Descriptors.FileDescriptor.InternalDescriptorAssigner in com.google.protobuf.Descriptors.FileDescriptor has been deprecated
[warn]     com.google.protobuf.Descriptors.FileDescriptor.InternalDescriptorAssigner assigner =
[warn] /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/grpc-example/grpc-example-java/hello-proxy-impl/target/scala-2.12/src_managed/main/example/myapp/helloworld/grpc/HelloWorldProto.java:44:1: com.google.protobuf.Descriptors.FileDescriptor.InternalDescriptorAssigner in com.google.protobuf.Descriptors.FileDescriptor has been deprecated
[warn]         new com.google.protobuf.Descriptors.FileDescriptor.    InternalDescriptorAssigner() {
[warn] /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/grpc-example/grpc-example-java/hello-proxy-impl/target/scala-2.12/src_managed/main/example/myapp/helloworld/grpc/HelloWorldProto.java:52:1: internalBuildGeneratedFileFrom(java.lang.String[],com.google.protobuf.Descriptors.FileDescriptor[],com.google.protobuf.Descriptors.FileDescriptor.InternalDescriptorAssigner) in com.google.protobuf.Descriptors.FileDescriptor has been deprecated
[warn]       .internalBuildGeneratedFileFrom(descriptorData,
[info] Done updating.
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 13 Java sources to /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/grpc-example/grpc-example-java/hello-impl/target/scala-2.12/classes ...
[warn] /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/grpc-example/grpc-example-java/hello-impl/target/scala-2.12/src_managed/main/example/myapp/helloworld/grpc/GreeterServiceClient.java:14:1: akka.stream.ActorMaterializer in akka.stream has been deprecated
[warn] import akka.stream.ActorMaterializer;
[warn] /Users/ignasi/git/projects/lightbend/lagom/lagom-samples/grpc-example/grpc-example-java/hello-impl/target/scala-2.12/src_managed/main/example/myapp/helloworld/grpc/GreeterServiceClientPowerApi.java:14:1: akka.stream.ActorMaterializer in akka.stream has been deprecated
```

So it boils down to `ActorMaterializer` and multiple methods or inner classes in `com.google.protobuf.Descriptors.FileDescriptor.*`.


I'm fine closing this PR.

Reply:
ActorMaterializer is deprecated and I can understand that it somewhat annoying to generate code with deprecation warnings. For that an option might be to have a a code generator setting for 2.5 or 2.6 (or detect that automatically from dependencies at generation time). Or simply branch off and maintain two.

Akka's internal protobuf v3 shouldn't be used by gRPC, so that should be unrelated.

FileDescriptor must be something else?

Reply:
> FileDescriptor must be something else?

I haven't dug into that one and it's not really a stopper (like you said "mewhat annoying to generate code with deprecation warnings"). Worth keeping an eye on it though.